front/src/applications/operationalStudies/helpers/TrainOpProjectionLazyLoader.ts

```
@@ -0,0 +1,116 @@
+import {
+  osrdEditoastApi,
+  type PostTrainScheduleOccupancyBlocksApiResponse,
+  type PostTrainScheduleProjectPathOpApiResponse,
+} from 'common/api/osrdEditoastApi';
+import type { TimetableItemId } from 'reducers/osrdconf/types';
+import {
+  extractEditoastIdFromPacedTrainId,
+  extractEditoastIdFromTrainScheduleId,
+  formatEditoastIdToTrainScheduleId,
+  isTrainScheduleId,
+} from 'utils/trainId';
+
+import TrainProjectionLazyLoaderAbstract, {
+  type ProjectionResult,
+  type TrainProjectionLazyLoaderOptions,
+} from './TrainProjectionLazyLoaderAbstract';
+
+export default class TrainOpProjectionLazyLoader extends TrainProjectionLazyLoaderAbstract {
+  readonly timetableItemIdProjected: TimetableItemId;
+
+  constructor(options: TrainProjectionLazyLoaderOptions, timetableIdProjected: TimetableItemId) {
+    super(options);
+    this.timetableItemIdProjected = timetableIdProjected;
+  }
+
+  async processBatch(batch: TimetableItemId[]) {
+    const { infraId, path, electricalProfileSetId } = this.options;
+    const { track_section_ranges } = path;
+
+    const { operational_points: operationalPoint } = await this.options
+      .dispatch(
+        osrdEditoastApi.endpoints.postInfraByInfraIdPathProperties.initiate({
+          infraId,
+          props: ['operational_points'],
+          pathPropertiesInput: {
+            track_section_ranges,
+          },
+        })
+      )
+      .unwrap();
+
+    const timetableItemProjectedOpIds: string[] = [];
+    const operationalPointDistances: number[] = [];
+
+    operationalPoint?.forEach(({ id, position }, index) => {
```

We should throw if operationalPoint is undefined I think (or at least return)

---

Done